File: algorithm/implimentation/knight/Knight.java
import java.util.Scanner;

public class Knight {
  public static void main(String[] args) {
    System.out.println("knight의 현위치 입력: ");
    Scanner sc = new Scanner(System.in);
    String cdn = sc.next();
    int x = cdn.charAt(0) - (int)'a' + 1;
    int y = Integer.parseInt(String.valueOf(cdn.charAt(1)));

    int[][] drt 
      = {{2, 1}, {2, -1}, {-2, 1}, {-2, -1}, {1, 2}, {-1, 2}, {1, -2}, {-1, -2}};
    int cnt = 0;

    for (int i = 0; i < 8; i++){
      int next_x, next_y;
      next_x = x + drt[i][0];
      next_y = y + drt[i][1];

      if (next_x > 0 && next_x < 9 && next_y > 0 && next_y < 9) cnt++;
    }

    System.out.println(cnt);
  }
}

File: algorithm/implimentation/knight/knight.py
# 나이트의 현위치 입력
cdn = list(input("나이트 현위치 입력: "))
x = ord(cdn[0]) - ord('a') + 1
y = int(cdn[1])

# 경우의 수
cnt = 0

# 이동할 수 있는 8가지 방향
drt = [(2, 1), (2, -1), (-2, 1), (-2, -1), (1, 2), (-1, 2), (1, -2), (-1, -2)]

# 경우의 수 세기
for d in drt:
  next_x = x + d[0]
  next_y = y + d[1]
  if next_x < 1 or next_x > 8 or next_y < 1 or next_y > 8:
    continue
  cnt += 1

print(cnt)